velox/expression/EvalCtx.cpp

```diff
@@ -274,16 +274,38 @@ void EvalCtx::ensureFieldLoaded(int32_t index, const SelectivityVector& rows) {
       baseRows->applyToSelected(
           [&rowNumbers](auto row) { rowNumbers.push_back(row); });
       rowSet = RowSet(rowNumbers);
+
+      // If we have a mapping that is not a single level of dictionary, we
+      // collapse this to a single level of dictionary. The reason is
+      // that the inner levels of dictionary will reference rows that
+      // are not loaded, since the load was done for only the rows that
+      // are reachable from 'field'.
+      if (field->encoding() != VectorEncoding::Simple::DICTIONARY ||
```

Comment:
should we have a unit test on EvalCtx as well to ensure we're doing the right thing with the input vector?